sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java

```diff
@@ -0,0 +1,186 @@
+        try {
+            Mono<PagedResponse<String>> pagedResponseMono = this.serviceClient.listRepositoriesSinglePageAsync(null, pageSize, context)
+                .map(res -> getPagedResponseWithContinuationToken(res));
+
+            return pagedResponseMono;
+
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    Mono<PagedResponse<String>> listRepositoriesNextSinglePageAsync(String nextLink, Context context) {
+        if (nextLink == null) {
+            return Mono.error(logger.logExceptionAsWarning(new IllegalArgumentException("nextLink parameter cannot be null.")));
+        }
+
+        try {
+            Mono<PagedResponse<String>> pagedResponseMono = this.serviceClient.listRepositoriesNextSinglePageAsync(nextLink, context);
+            return pagedResponseMono.map(res -> getPagedResponseWithContinuationToken(res));
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+
+    /**
+     * Delete the repository identified by `name`.
+     *
+     * @param name Name of the image (including the namespace).
+     * @return deleted repository.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DeleteRepositoryResult>> deleteRepositoryWithResponse(String name) {
+        return withContext(context -> deleteRepositoryWithResponse(name, context));
+    }
+
+    Mono<Response<DeleteRepositoryResult>> deleteRepositoryWithResponse(String name, Context context) {
+        if (CoreUtils.isNullOrEmpty(name)) {
+            throw logger.logExceptionAsError(new NullPointerException("repository name can't be null or empty."));
+        }
+
+        return this.serviceClient.deleteRepositoryWithResponseAsync(name, context);
+    }
+
+    /**
+     * Delete the repository identified by `name`.
+     *
+     * @param name Name of the image (including the namespace).
+     * @return deleted repository.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DeleteRepositoryResult> deleteRepository(String name) {
+        return withContext(context -> this.deleteRepository(name, context));
+    }
+
+    Mono<DeleteRepositoryResult> deleteRepository(String name, Context context) {
+        if (CoreUtils.isNullOrEmpty(name)) {
+            throw logger.logExceptionAsError(new NullPointerException("repository name can't be null or empty."));
```

Comment:
Let's return a `Mono.error`

Reply:
I actually got this from one of the code sanity checks - may be we should also update it to give this message instead? Not sure how easy\difficult is it to do it.

---
In reply to: [602630773](https://github.com/Azure/azure-sdk-for-java/pull/20139#discussion_r602630773) [](ancestors = 602630773)